cmake/Dependencies.cmake

```diff
@@ -84,14 +84,12 @@ if(USE_ROCM)
     list(APPEND HIP_CXX_FLAGS -Wno-unused-command-line-argument)
     list(APPEND HIP_CXX_FLAGS -Wno-duplicate-decl-specifier)
     list(APPEND HIP_CXX_FLAGS -DUSE_MIOPEN)
-
-    set(HIP_HCC_FLAGS ${HIP_CXX_FLAGS})
+    list(APPEND HIP_CXX_FLAGS -fno-gpu-rdc)
+    list(APPEND HIP_CXX_FLAGS -Wno-defaulted-function-deleted)
     # Ask hcc to generate device code during compilation so we can use
     # host linker to link.
-    list(APPEND HIP_HCC_FLAGS -fno-gpu-rdc)
-    list(APPEND HIP_HCC_FLAGS -Wno-defaulted-function-deleted)
     foreach(gloo_rocm_arch ${GLOO_ROCM_ARCH})
-      list(APPEND HIP_HCC_FLAGS --amdgpu-target=${gloo_rocm_arch})
+        list(APPEND HIP_CXX_FLAGS --amdgpu-target=${gloo_rocm_arch})
```

Comment:
Is HIP_HCC_FLAGS specifically for hcc compiler? The fix is that we'll be using HIP_CXX_FLAGS explicitly when we trigger hcc, is that correct?

Reply:
good catch! I am in the middle of refactoring, we should be using HIP_HCC_FLAGS. thx